pet-shop/src/Components/Animals.jsx

```diff
@@ -1,28 +1,25 @@
 import React, { useState, useEffect } from 'react';
 
-export default function Animals(){
-
-    const [animals, setAnimals] = useState([]);
+export default function Animals({animals, setAnimals, originalAnimals, setOriginalAnimals}){  
 
     useEffect(() => {
         fetch('http://localhost:8000/api/fetch')
            .then((response) => response.json())
            .then((data) => {
               console.log(data);
               setAnimals(data.result);
+              setOriginalAnimals(data.result);
            })
            .catch((err) => {
               console.log(err.message);
-           });
-           // New endpoint for the total request
+           });      
```

Comment:
This isn't a major issue but I feel like now that `const [animals, setAnimals] = useState([]);` is a level above it would make sense to move the initial fetch request to be on that same level.

Also completely random but I noticed the request was being made twice, which was very confusing so I did some research and found this: https://stackoverflow.com/a/60619061 
"StrictMode renders components twice (on dev but not production) in order to detect any problems with your code and warn you about them (which can be quite useful)."

How bizarre but useful to know I suppose. Strict mode is enabled by this code in main.jsx `<React.StrictMode>`